packages/core/src/CML/coreToCml/coreToCml.ts

```diff
@@ -436,17 +448,42 @@ export const txWitnessBootstrap = (
   return witnesses;
 };
 
-export const witnessSet = (scope: ManagedFreeableScope, witness: Cardano.Witness): TransactionWitnessSet => {
-  const cslWitnessSet = scope.manage(TransactionWitnessSet.new());
-  const vkeyWitnesses = scope.manage(Vkeywitnesses.new());
+const mapRedeemerTag = (scope: ManagedFreeableScope) => ({
+  [RedeemerPurpose.spend]: scope.manage(RedeemerTag.new_spend()),
```

Comment:
This currently creates objects that are never used
```suggestion
const mapRedeemerTag = (scope: ManagedFreeableScope, redeemerPurpose: RedeemerPurpose) => {
  switch (redeemerPurpose) {
    case RedeemerPurpose.spend: return scope.manage(RedeemerTag.new_spend());
```

Reply:
Right, the idea was to get rid of the switch case but it seems that's the way to manage the obj properly

Reply:
d6aee0be84cc47170a0942ac940bdc473fca4591 @mkazlauskas